geode-core/src/integrationTest/java/org/apache/geode/internal/cache/eviction/EvictionIntegrationTest.java

```diff
@@ -0,0 +1,190 @@
+
+
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(CategoryWithParameterizedRunnerFactory.class)
+@Category({EvictionTest.class})
+public class EvictionIntegrationTest {
+
+  @Parameterized.Parameters(name = "offHeap={0}")
+  public static Collection booleans() {
+    return Arrays.asList(true, false);
+  }
+
+  @Parameterized.Parameter
+  public boolean offHeap;
+
+  @Rule
+  public ServerStarterRule server = new ServerStarterRule().withNoCacheServer();
+
+  @Before
+  public void setUp() throws Exception {
+    if (offHeap) {
+      server.withProperty(OFF_HEAP_MEMORY_SIZE, "200m");
+    }
+    server.startServer();
+  }
+
+  @Test
+  public void testEntryLruEvictions() {
+    int maxEntry = 3;
+    PartitionedRegion pr1 = (PartitionedRegion) server.createPartitionRegion("PR1",
+        f -> f.setOffHeap(offHeap)
+            .setEvictionAttributes(
+                EvictionAttributes.createLRUEntryAttributes(maxEntry,
+                    EvictionAction.LOCAL_DESTROY)),
+        f -> f.setRedundantCopies(0).setTotalNumBuckets(4));
+
+    // put in one extra entry after maxEntry
+    for (int counter = 1; counter <= maxEntry + 1; counter++) {
+      pr1.put(counter, new byte[1024 * 1024]);
+    }
+
+    assertThat(pr1.getTotalEvictions()).isEqualTo(1);
+  }
+
+  @Test
+  public void testEntryLru() {
+    int maxEntry = 12;
+    PartitionedRegion pr1 = (PartitionedRegion) server.createPartitionRegion("PR1",
+        f -> f.setOffHeap(offHeap)
+            .setEvictionAttributes(
+                EvictionAttributes.createLRUEntryAttributes(maxEntry,
+                    EvictionAction.LOCAL_DESTROY)),
+        f -> f.setRedundantCopies(0).setTotalNumBuckets(4));
+
+    for (int i = 0; i < 3; i++) {
+      // assume mod-based hashing for bucket creation
+      pr1.put(i, "value0");
+      pr1.put(i + pr1.getPartitionAttributes().getTotalNumBuckets(), "value1");
+      pr1.put(i + (pr1.getPartitionAttributes().getTotalNumBuckets() * 2), "value2");
+    }
+    pr1.put(3, "value0");
+
+    for (int i = 0; i < 2; i++) {
+      pr1.put(i + pr1.getPartitionAttributes().getTotalNumBuckets() * 3, "value1");
+    }
+    assertThat(pr1.getTotalEvictions()).isEqualTo(0);
+  }
+
+  @Test
+  public void testCheckEntryLruEvictionsIn1DataStore() {
+    int extraEntries = 10;
+    int maxEntries = 20;
+    PartitionedRegion pr1 = (PartitionedRegion) server.createPartitionRegion("PR1",
+        f -> f.setOffHeap(offHeap)
+            .setEvictionAttributes(
+                EvictionAttributes.createLRUEntryAttributes(maxEntries,
+                    EvictionAction.LOCAL_DESTROY)),
+        f -> f.setRedundantCopies(1).setTotalNumBuckets(5));
+
+    for (int counter = 1; counter <= maxEntries + extraEntries; counter++) {
+      pr1.put(counter, new byte[1024 * 1024]);
+    }
+
+    assertThat(pr1.getTotalEvictions()).isEqualTo(extraEntries);
+
+    for (final Iterator i = pr1.getDataStore().getAllLocalBuckets().iterator(); i.hasNext();) {
```

Comment:
There are a handful of places in the tests where, for instance, IntelliJ's inspections can identify where the old iterator style is used where the more readable `for each` loop style could be used instead.  A minor thing, but as long as the focus is refactoring, it's an easy thing to let your IDE do.